hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/ReactiveOuterJoinLoader.java

```diff
@@ -109,184 +102,160 @@ public ReactiveOuterJoinLoader(SessionFactoryImplementor factory, LoadQueryInflu
-								"could not load an entity batch: " + MessageHelper.infoString(
-										getEntityPersisters()[0],
-										queryParameters.getOptionalId(),
-										session.getFactory()
-								),
-								sql
-						);
-					}
-				});
-	}
-
-	/**
-	 * Resolve any IDs for currently loaded objects, duplications within the
-	 * <tt>ResultSet</tt>, etc. Instantiate empty objects to be initialized from the
-	 * <tt>ResultSet</tt>. Return an array of objects (a row of results) and an
-	 * array of booleans (by side-effect) that determine whether the corresponding
-	 * object should be initialized.
-	 */
-	private CompletionStage<Object[]> getReactiveRow(
-			final ResultSet rs,
-			final Loadable[] persisters,
-			final EntityKey[] keys,
-			final QueryParameters queryParameters,
-			final EntityKey optionalObjectKey,
-			final LockMode[] lockModes,
-			final List hydratedObjects,
-			final SharedSessionContractImplementor session) throws HibernateException, SQLException {
-		final int cols = persisters.length;
-		final EntityAliases[] entityAliases = getEntityAliases();
-
-		if ( LOG.isDebugEnabled() ) {
-			LOG.debugf( "Result row: %s", StringHelper.toString( keys ) );
-		}
-
-		CompletionStage<Object[]> rowResultsStage = CompletionStages.completedFuture( new Object[cols] );
-
-		for ( int i = 0; i < cols; i++ ) {
-			final int current = i;
-			rowResultsStage = rowResultsStage.thenCompose( rowResults -> {
-				try {
-					return getObject(rs, persisters, keys, queryParameters.getOptionalObject(), optionalObjectKey, lockModes, hydratedObjects, session, entityAliases, current)
-							.thenApply( obj -> {
-								rowResults[current] = obj;
-								return rowResults;
-							} );
-				}
-				catch (SQLException sqle) {
-					throw getFactory().getJdbcServices().getSqlExceptionHelper().convert(
-							sqle,
-							"could not load an entity batch: " + MessageHelper.infoString(
-									getEntityPersisters()[0],
-									queryParameters.getOptionalId(),
-									session.getFactory()
-							),
-							sql
-					);
-				}
-			});
-		}
-
-		return rowResultsStage;
-	}
-
-	private CompletionStage<Object> getObject(ResultSet rs, Loadable[] persisters, EntityKey[] keys, Object optionalObject, EntityKey optionalObjectKey, LockMode[] lockModes, List hydratedObjects, SharedSessionContractImplementor session, EntityAliases[] entityAliases, int i) throws SQLException {
-		Object object = null;
-		EntityKey key = keys[i];
-		if ( keys[i] != null ) {
-			//If the object is already loaded, return the loaded one
-			object = session.getEntityUsingInterceptor( key );
-			if ( object != null ) {
-				instanceAlreadyLoaded( rs, i, persisters[i], key, object, lockModes[i], hydratedObjects, session);
-			}
-			else {
-				object = instanceNotYetLoaded( rs, i, persisters[i], entityAliases[i].getRowIdAlias(), key, lockModes[i],
-						optionalObjectKey, optionalObject, hydratedObjects, session );
-			}
-		}
-		return CompletionStages.completedFuture( object );
-	}
```

Comment:
@DavideD none of the above code was ever called.

Do you know the story behind it? Can we just delete it?

Reply:
What do you mean it's not used? It should be called by the class. If it's not needed, it's probably because there's a test missing. It's the equivalent of `Loader#getRow`.

Reply:
It was the result of converting the `OuterJoinLoader`, so it's possible that I've not added a test for that particular method. Not sure when we expect it to be called.

Reply:
No, none of that code is ever executed. I believe it's code that in Hibernate ORM is related to processing scrollable result sets. (But I'm not sure.)

Reply:
(I mean, it's never executed, because there is nothing in the codebase that references it, not because of missing tests.)

Reply:
Ah, sorry. I just noticed that it's not getObject that doesn't get called but collectionPersister.  I think it was part of that big PR regarding CollectionPersister that I've changed and when I've cleaned it up I didn't realized that these methods were never called.

Reply:
Yes, that's what I figured.

Reply:
Yes, they were called by `rxProcessResultSet` that was an adaptation of `processResultSet`. Sorry about that.